src/witan/cic/episodes.clj

```diff
@@ -554,35 +581,131 @@
 (def create-header-xf
   (map create-header))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Separate episodes for removal for further processing
+(defn separate-episodes-for-removal[[id {::keys [ssda903-episodes] :as rec}]]
+  (let [episodes         (x/into []
+                                 (remove tagged-for-removal?)
+                                 ssda903-episodes)
+        removed-episodes (x/into []
+                                 (filter tagged-for-removal?)
+                                 ssda903-episodes)]
```

Comment:
would group-by or partition by make more sense? Not sure about computation time, but it would be a single step rather than two. Just a consideration

Reply:
hmm... I'd have to rename the keys after creating the maps, so still two steps. atm, ::removed-episodes is only there so that I can debug things. We're only really interested in the scrubbed episodes after this, so if I was going to remove something I'd like it to be the removed-episodes work entirely.